library/spdm_requester_lib/libspdm_req_challenge.c

```diff
@@ -113,15 +113,19 @@ return_status try_spdm_challenge(IN void *context, IN uint8_t slot_id,
 
     status = spdm_send_spdm_request(spdm_context, NULL,
                     sizeof(spdm_request), &spdm_request);
-    if (RETURN_ERROR(status)) {
+    if (status == RETURN_TIMEOUT) {
+        return status;
+    } else if (RETURN_ERROR(status)) {
         return RETURN_DEVICE_ERROR;
     }
 
```

Comment:
This file, and lots of other requests, needs `spdm_context->crypto_request = TRUE;`

Reply:
spdm_context->crypto_request is set to TRUE in libspdm_init_context(). To GET_CAPABILITIES , GET_VERSION , or NEGOTIATE_ALGORITHMS request message, it is set to FALSE.

Reply:
How does the sequence `NEGOTIATE_ALGORITHMS (crypto_request == FALSE)` -> `GET_DIGESTS (crypto_request == TRUE)` occur then? There's no `libspdm_init_context` in between the two requests.

Reply:
I got your point. Will update the change  for spdm_context->crypto_request = TRUE.